### PR TITLE
1061 refactor core file validators based on aind data schema models expected modalities

### PR DESCRIPTION
**Pull request by dbirman**
This PR takes advantage of the new `ExpectedFiles` model in `aind-data-schema-models` to refactor the core metdata validator to check for required/excluded files for all modalities. Currently the validators only check for ecephys/smartspim platform to confirm they have a few required files. The code now checks by modality (not platform) and confirms that all expected/required files are included.

In the event that two modalities are in conflict the code prioritizes REQUIRED > OPTIONAL > EXCLUDED

### CODE REVIEW

**Comment by dbirman** [comment]
There's some QC file changes that snuck into this branch... I'll remove those

**Comment by dbirman** [comment]
There's a bug in this PR -- it can't handle modalities with a '-' in the name (need to .replace('-','_'))

**Comment by jtyoung84** [comment]
> There's a bug in this PR -- it can't handle modalities with a '-' in the name (need to .replace('-','_'))

I wonder if it'd be helpful to add the modality folder name explicitly in https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/src/aind_data_schema_models/models/modalities.csv

**Comment by dbirman** [comment]
> > There's a bug in this PR -- it can't handle modalities with a '-' in the name (need to .replace('-','_'))
> 
> ~I wonder if it'd be helpful to add the modality folder name explicitly in https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/src/aind_data_schema_models/models/modalities.csv~
> 
> I guess I'm just using the abbreviation

Yeah I think we only ever use the abbreviation everywhere

File: src/aind_data_schema/core/metadata.py [hunk 1]
```diff
@@ -189,17 +195,43 @@ def validate_metadata(self):
...
+                        file_requirement == FileRequirement.OPTIONAL
+                        and requirement_dict[file] == FileRequirement.EXCLUDED
+                    ):
+                        # override, required wins over all else, and optional wins over excluded
+                        requirement_dict[file] = file_requirement
+
+            for file in CORE_FILES:
+                # Check required case
+                if requirement_dict[file] == FileRequirement.REQUIRED and not getattr(self, file):
+                    raise ValueError(f"{modality.abbreviation} metadata missing required file: {file}")
```
Comment (jtyoung84) [inline src/aind_data_schema/core/metadata.py:224]: This `modality.abbreviation` raises a warning about `referencing before assignment`. Does the modality.abbreviation need to be tracked in the `requirement_dict` in the previous loop?

Comment (dbirman) [reply src/aind_data_schema/core/metadata.py]: fixed in https://github.com/AllenNeuralDynamics/aind-data-schema/pull/1084/commits/529bf44000cb35c0e1d673e5b4f3b559e2ca60c7

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**